From: David Ahern <dsahern@gmail.com>
Date: Sat Sep 28 14:28:00 2013 -0600
Subject: perf stat: Add units to nanosec-based counters
Git-commit: 4bbe5a61f29b13437a6a16467328d3bae8fce9e7
Patch-mainline: v3.13-rc1
References: FATE#318981, FATE#318982, FATE#318983, FATE#318984
Signed-off-by: Tony Jones <tonyj@suse.de>

    perf stat: Add units to nanosec-based counters
    
    Ingo pointed out that the task-clock counter should have the units
    explicitly stated since it is not a counter.
    
    Before:
    
    perf stat -a -- sleep 1
    
     Performance counter stats for 'sleep 1':
    
          16186.874834 task-clock          #   16.154 CPUs utilized
    ...
    
    After:
    
    perf stat -a -- sleep 1
    
     Performance counter stats for 'system wide':
    
          16146.402138 task-clock (msec)   #   16.125 CPUs utilized
    ...
    
    Reported-by: Ingo Molnar <mingo@kernel.org>
    Signed-off-by: David Ahern <dsahern@gmail.com>
    Cc: Frederic Weisbecker <fweisbec@gmail.com>
    Cc: Ingo Molnar <mingo@kernel.org>
    Cc: Jiri Olsa <jolsa@redhat.com>
    Cc: Namhyung Kim <namhyung.kim@lge.com>
    Cc: Peter Zijlstra <a.p.zijlstra@chello.nl>
    Cc: Stephane Eranian <eranian@google.com>
    Link: http://lkml.kernel.org/r/1380400080-9211-4-git-send-email-dsahern@gmail.com
    Signed-off-by: Arnaldo Carvalho de Melo <acme@redhat.com>

diff --git a/tools/perf/builtin-stat.c b/tools/perf/builtin-stat.c
index 2178e66..1a9c95d 100644
--- a/tools/perf/builtin-stat.c
+++ b/tools/perf/builtin-stat.c
@@ -706,10 +706,13 @@ static void nsec_printout(int cpu, int nr, struct perf_evsel *evsel, double avg)
 {
 	double msecs = avg / 1e6;
 	const char *fmt = csv_output ? "%.6f%s%s" : "%18.6f%s%-25s";
+	char name[25];
 
 	aggr_printout(evsel, cpu, nr);
 
-	fprintf(output, fmt, msecs, csv_sep, perf_evsel__name(evsel));
+	scnprintf(name, sizeof(name), "%s%s",
+		  perf_evsel__name(evsel), csv_output ? "" : " (msec)");
+	fprintf(output, fmt, msecs, csv_sep, name);
 
 	if (evsel->cgrp)
 		fprintf(output, "%s%s", csv_sep, evsel->cgrp->name);
